src/dirext.rs

```diff
@@ -737,18 +765,9 @@ impl CapStdExtDirExt for Dir {
         perms: cap_std::fs::Permissions,
     ) -> Result<()> {
         self.atomic_replace_with(destname, |f| -> io::Result<_> {
-            // If the user is overriding the permissions, let's make the default be
-            // writable by us but not readable by anyone else, in case it has
-            // secret data.
-            #[cfg(unix)]
-            {
-                use cap_std::fs::PermissionsExt;
-                let perms = cap_std::fs::Permissions::from_mode(0o600);
-                f.get_mut().as_file_mut().set_permissions(perms)?;
-            }
+            // File is already open (with mode 0o666 | umask), set the new permissions right away
```

Comment:
Again this change is unrelated to atomicity as far as I can tell. If you want to change our permissions handling please do so in another PR.